pkg/mock/deployment.go

```diff
@@ -0,0 +1,37 @@
+package mock
+
+import (
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/Meetic/blackbeard/pkg/resource"
+)
+
+type deploymentRepository struct {
+	kubernetes.Interface
+}
+
+func NewDeploymentRepository(kubernetes kubernetes.Interface) resource.DeploymentRepository {
+	return &deploymentRepository{kubernetes}
+}
+
+func (deploymentRepository) List(namespace string) (resource.Deployments, error) {
+	if namespace == "testko" {
```

Comment:
Instead of using a condition on a namespace value maybe you should create 2 different NewDeploymentRepository. One that return a deployment in progress the other one ready